File: backend/models/Goal.js
const mongoose = require("mongoose");

const contributionSchema = new mongoose.Schema({
    contributor: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Person"
    },
    amount: {
        type: Number
    },
    createdAt: {
        type: Date,
        default: Date.now
    }
}, {
    versionKey: false
})

const goalSchema = new mongoose.Schema({
    name: { type: String },
    amount: { type: Number },
    dueDate: { type: String },
    contributions: [contributionSchema],
    image: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Image"
    }
}, {
    versionKey: false
})

module.exports = mongoose.model("Goal", goalSchema);

File: frontend/app/src/main/java/com/componentes/vaquita/domain/model/Goal.kt
package com.componentes.vaquita.domain.model

import com.google.gson.annotations.SerializedName


data class Goal(val _id: String?,
                @SerializedName("name") var name: String?,
                var amount: Int?,
                var dueDate: String?,
                var contributions: MutableList<Contribution>?,
                var image: String?)